OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -149,17 +149,14 @@ protected virtual bool CanAttack(Actor self, in Target target)
 			if (!target.IsValidFor(self))
 				return false;
 
-			if (!HasAnyValidWeapons(target))
+			if (!HasAnyValidWeapons(target, checkForReloadingState: true))
```

Comment:
Style nit: IMO its not immediately clear what `checkForReloadingState` means, without reading the implementation - specifically the "state" in the name makes me think that this might be doing something more complicated than it really is. How about something like `requireLoaded` or `reloadingIsInvalid`?